pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1848,7 +1848,9 @@ static __always_inline int do_should_trace(event_data_t *data)
         bool filter_out = (config & FILTER_PID_OUT) == FILTER_PID_OUT;
         u64 max = data->config->pid_max;
         u64 min = data->config->pid_min;
-        if (!uint_filter_matches(filter_out, &pid_filter, context->host_tid, max, min))
+        // the user might have given us a tid - check for it too
+        if ((!uint_filter_matches(filter_out, &pid_filter, context->host_pid, max, min)) &&
+            (!uint_filter_matches(filter_out, &pid_filter, context->host_tid, max, min)))
```

Comment:
I thought about it too, but this means we will need to:
1. Create a new flag for tid
2. Create a new filter map
3. Add more instructions to the do_should_trace function

I also see an advantage of letting the user only specifying one of the threads of a process in order to trace the whole process using the `--trace pid` flag.

I'm open to get more opinions here though, and if you think otherwise, we can go with that direction.
@rafaeldtinoco @josedonizetti @geyslan WDYT?

Reply:
> I'm open to get more opinions here though, and if you think otherwise, we can go with that direction.
@rafaeldtinoco @josedonizetti @geyslan WDYT?

Shouldn't it be better to first check if tid and pid are different, avoiding two lookups?

Reply:
Can we have pid/tid flags but use the same ebpf infrastructure just like above? Instead of creating a new map and changing the `do_should_trace`, it would only be a semantic change on the userland but the same implementation below?

Reply:
> Shouldn't it be better to first check if tid and pid are different, avoiding two lookups?

That's not a big issue here since we cache the result of the do_should_trace() function

Reply:
> Can we have pid/tid flags but use the same ebpf infrastructure just like above? Instead of creating a new map and changing the `do_should_trace`, it would only be a semantic change on the userland but the same implementation below?

That can be a good idea now, but since we plan to support multiscope filters in the future, it can become problematic for scopes that use the pid/tid filters differently between them